vuu-ui/packages/vuu-shell/src/application-settings/ApplicationSettingsPanel.tsx

```diff
@@ -1,58 +1,193 @@
+    // Switch for booleans
+    if (values?.length == 2) {
+      if (property.type === "boolean") {
+        return (
+          <Switch
+            label={property.label}
+            value={currentValue}
+            onChange={changeHandler}
+          ></Switch>
+        );
+      }
+    }
+    // Toggle Box for 1 or 2 values
+    if (values?.length <= 2) {
+      return (
+        <ToggleButtonGroup value={currentValue} onChange={changeHandler}>
+          {values?.map((value) => (
+            <ToggleButton key={value} value={value}>
+              {value}
+            </ToggleButton>
+          ))}
+        </ToggleButtonGroup>
+      );
+
+      // Dropdown for more than 2 values provided
+    } else if (values?.length > 2) {
+      return (
+        <Dropdown value={currentValue} onSelectionChange={selectHandler}>
+          {values?.map((value) => {
+            if (typeof value === "object") {
+              return (
+                <Option
+                  value={value.label}
+                  key={value.value}
+                  data-field={property.name}
+                ></Option>
+              );
+            } else {
+              return (
+                <Option
+                  value={value}
+                  key={value}
+                  data-field={property.name}
+                ></Option>
+              );
+            }
+          })}
+        </Dropdown>
+      );
+    }
+  } else {
+    return <Input></Input>;
+  }
+}
 
-export interface ApplicationSettingsPanelProps
+// Props for Panel
+export interface ApplicatonSettingsPanelProps
   extends HTMLAttributes<HTMLDivElement> {
-  settings: { [key: string]: unknown };
+  applicationSettingsSchema: SettingsSchema;
+  applicationSettings: Record<string, string | number | boolean>;
+  onApplicationSettingChanged: (
+    propertyName: string,
+    value: string | number | boolean
+  ) => void;
 }
 
+// Gets the field form element for a dropdown selection box for selection change handler
+export const queryFieldFormElement = <T extends HTMLElement = HTMLElement>(
+  el: HTMLElement | EventTarget | null
+) => {
+  if (el === null) {
+    return null;
+  }
+  const dataFieldValue = el.getAttribute("data-field");
+  const saltFormField = document.querySelector(".saltFormField") as HTMLElement;
+  if (saltFormField && saltFormField.dataset.field) {
+    return document.querySelector(`[data-field="${dataFieldValue}"]`);
+  }
+};
+
+// Generates application settings form component
+export const SettingsForm = ({
+  applicationSettingsSchema,
+  applicationSettings,
+  onApplicationSettingChanged,
+}: ApplicatonSettingsPanelProps) => {
+  // Settings Handler for toggle and input buttons
+  const onSettingChanged = useCallback<FormEventHandler>((event) => {
+    const fieldElement = queryClosest(event.target, "[data-field]");
+    const fieldName = getFieldName(fieldElement);
+    onApplicationSettingChanged(fieldName, event.target.value);
+  }, []);
```

Comment:
must not miss the dependencies from the dependencyArray. If you have the ESLint extension installed in VSCode, it should be warning you when these are missing